staging/src/k8s.io/cli-runtime/pkg/resource/fallback_query_param_verifier.go

```diff
@@ -44,12 +43,13 @@ func NewFallbackQueryParamVerifier(primary Verifier, secondary Verifier) Verifie
 // HasSupport returns an error if the passed GVK does not support the
 // query param (fieldValidation), as determined by the primary and
 // secondary OpenAPI endpoints. The primary endoint is checked first,
-// but if it not found, the secondary attempts to determine support.
-// If the GVK supports the query param, nil is returned.
+// but if there is an error retrieving the OpenAPI V3 document, the
+// secondary attempts to determine support. If the GVK supports the query param,
+// nil is returned.
 func (f *fallbackQueryParamVerifier) HasSupport(gvk schema.GroupVersionKind) error {
 	err := f.primary.HasSupport(gvk)
-	if errors.IsNotFound(err) {
-		klog.V(7).Infoln("openapi v3 endpoint not found...falling back to legacy")
+	if !IsParamUnsupportedError(err) {
```

Comment:
Don't you need to check if err is != nil here first? if it doesn't crash in the test it's probably fine. But still that logic is a little strange.

Reply:
The err == nil check happens inside the `IsParamUnsupportedError()` function, but I can make it more readable by adding that check here as well.

Reply:
What I find confusing that way is that if "err == nil", then I'm guessing it is NOT a `ParamUnsupportedError`, but that will go inside the if condition when it shouldn't?

IOW,
- `IsParamUnsupportedError(err)` => `err != nil && err.(*paramUnsupportedError)`
- `!IsParamUnsupportedError(err)` => `err == nil || !err.(*paramUnsupportedError)`
- What we actually want is `err != nil && !err.(*paramUnsupportedError)`

Reply:
OK. I think this is correct now. We should only check the secondary endpoint on the following two conditions:
1. The primary determined the GVK supports the query param (`nil` is returned by `primary.HasSupport`).
2. The primary determined the GVK does **not** support the query param (`paramUnsupportedError` is returned by `primary.HasSupport`).

I have added to tests to ensure the secondary endpoint is **not** called when the previous two conditions happen.

Reply:
> We should only check the secondary endpoint on the following two conditions

I think you accidentally said the opposite of what you wanted to say, but yeah, that's correct, thank you!